assets/scripts/admin/surveysettings.js

```diff
@@ -158,29 +158,38 @@ function guidGenerator() {
 function validateSettingsForm($form) {
     switch ($form.attr('id')) {
         case 'publication':
-            return validateEndDate();
+            return validatePublicationForm();
         default:
             return true;
     }
 }
 
 /**
- * Validates that the expiration date is not lower than the start date
+ * Validates the "Publication & access" form
  */
-function validateEndDate() {
+ function validatePublicationForm() {
```

Comment:
That's great, _but_ I have one more complaint: validatePublicationForm is not needed. Instead, just using function composition:

```js
return validateEndDateHigherThanStart(
  $('#startdate_datetimepicker').data('DateTimePicker'),
  $('#expires_datetimepicker').data('DateTimePicker'),
  expirationLowerThanStartError
);
```

Now you have composable parts working together!